### PR TITLE
Publish daemon_lib to sonatype instead of repository.liveramp.com

### PR DESCRIPTION
**Pull request by joshk0**
Add deploy script and credentials to sonatype that are usable from
master builds using a 'publish' profile. PR builds will use only
the regular profile.

### CODE REVIEW

**Comment by joshk0** [comment]
Hmm... Does it make sense for the entire publish profile to go into LiveRamp/pom-common ? Which would have the deploy script as a resource. Then to Sonatype something, you would just need to add a .travis.yml with the encrypted secrets, .travis/mvn-settings.xml and .travis/gpg.asc.enc.

Rationale: pom-common itself needs to publish to Sonatype anyway so why copy unnecessary things?

**Comment by bpodgursky** [comment]
While it would reduce copy-pasting a bit to have this in pom-common, I would still prefer to move towards removing pom-common entirely and having OSS projects work standalone.   Dealing with pom-common versioning on snapshots vs releases is a nuisance and I'd like to err towards our OSS projects being as easy to consume as possible.

**Comment by joshk0** [comment]
OK, any feedback on the way I've done it here? It's important as, given your advice, this would be the prototype to follow in all other OSS repos.

**Comment by bpodgursky** [comment]
What exactly is the workflow that triggers a deploy on travis?  I see that in the deploy script you have "-DskipTests=true" and I'm trying to figure out if this could end up skipping tests if someone pushes to master.  If possible, I'd rather run the tests before every deploy.

**Comment by joshk0** [comment]
It's my belief that the deploy step runs immediately after the testing phase but only on master builds. As a result, removing skipTests would result in two runs of tests on master builds, but you're right that I'm not 100% confident.

Happy to have a look at the resulting master build after merging this.

File: README.md [hunk 1]
```diff
@@ -20,9 +20,12 @@ In Maven, make your project section look like this:
 
   <repositories>
     <repository>
-      <id>liveramp-repositories</id>
-      <name>Liveramp Repositories</name>
-      <url>http://repository.liveramp.com/artifactory/liveramp-repositories</url>
+      <id>maven-snapshots</id>
+      <url>http://oss.sonatype.org/content/repositories/snapshots</url>
```
Comment (bpodgursky) [inline README.md:24]: Once we do a proper release to central, we should update this section to reference it instead.

**Comment by bpodgursky** [comment]
Looks pretty good to me otherwise.  Nice.

File: README.md (hunk 1 above)
Comment (joshk0) [reply README.md:24]: yep, will do -- think there's a separate and slightly longer discussion about versioning to figure out here, so prefer to leave it as-is for now.

**Comment by joshk0** [comment]
[check it out](https://travis-ci.org/LiveRamp/daemon_lib/builds/513893410)

* `mvn test -B` runs first
* then the `deploy -Ppublish` runs after as a deploy step
* so: tests run exactly once

**Comment by bpodgursky** [comment]
nice!

**Pull request merged**